### ISSUE TITLE
Custom props in CSF story format

### ISSUE THREAD

**Issue by kushalmahajan** [storybookjs/storybook#10018]
I use typescript and it would be nice to have custom props to add instead of automatic parsing. 

Here 

```
export default {
	title: "Avatar",
	component: Avatar,
	parameters: {
		componentSubtitle: "Size",
                 props: ["size", "type", ....]
	}
};
```

There are some cases where we merge `React.HTMLAttributes` and custom props for a component. 

Example - 

```
export type NativeButtonProps = {
	onClick: React.MouseEventHandler<HTMLElement>;
	htmlType?: ButtonHTMLType;
} & BaseButtonProps &
	Omit<React.ButtonHTMLAttributes<unknown>, "type" | "onClick">;

type ButtonProps = Partial<AnchorButtonProps & NativeButtonProps>;
export const Button: React.FC<ButtonProps> = (props: ButtonProps) => {
```
In such a case, earlier in story API format, I was handling this via my custom TableComponent just for that specific component story. However, can't see any options for in CSF format. 

I do have a workaround for it. If I use `React.propTypes` in addition to it, I have the desired outcome but it defeats the purpose of having Typescript for props.

**Comment by shilman** [comment]
@kushalmahajan we'll have a better option for you in SB6 as part of this effort

https://docs.google.com/document/d/1Mhp1UFRCKCsN8pjlfPdz8ZdisgjNXeMXpXvGoALjxYM/edit?usp=sharing

**Comment by kushalmahajan** [comment]
Ok. Will wait for it

**Comment by shilman** [comment]
Args-based props tables are available in 6.0-alpha, with more configuration options

https://gist.github.com/shilman/69c1dd41a466bae137cc88bd2c6ef487

**Comment by kushalmahajan** [comment]
This is great! Got more in this release. Read the examples. Will try out soon.
